mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -433,6 +433,34 @@ def GPU_GPUFuncOp : GPU_Op<"func", [
   let hasVerifier = 1;
 }
 
+def GPU_DynamicSharedMemoryOp : GPU_Op<"dynamic_shared_memory", 
+  [ParentOneOf<["GPUFuncOp", "LaunchOp", 
+                "mlir::LLVM::LLVMFuncOp", "mlir::func::FuncOp"]>]>
```

Comment:
The verifier seems a bit overkill to me: I should be able to write my own kernel function I think?


Also this looks like it should be an operation without side-effects, but then I should also be able to sink it which would break the "getParent" here.

Can we get away with this verifier check and make the client code safe instead?

Reply:
We actually need an Op that has SymbolTable, so I added this check in the verifier and removed this check.
I also added `Pure`, yes the op has no memory side effect and can be hoisted.